### ISSUE TITLE
PHP Action Scheduler requests (/wp-admin/admin-ajax.php?action=as_async_request_queue_runner) take a long time to complete

### ISSUE THREAD

**Issue by BradFD** [wp-media/wp-rocket#6662]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
- Yes
- Used the search feature to ensure that the bug hasn’t been reported before
- Yes

**Describe the bug**
PHP 8.3, WP Rocket 3.16 - Activate Preloading is OFF,  RUCSS is ON. 4vCPU server is always under 25% CPU load, RAM is only ~30% used, Server is well sized for load and not stressed.  

Looking at my PHP access logs the log show that every time the following action schedular related request comes up it takes around 5-6 seconds to complete. 
Log entry split below to show emphasis, timing %d variable in seconds I added to log file:
"POST /wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe" 200 0 - 366682 425175 
timing in second add (see steps to reproduce )  ---> 5.132 
14680064 1.95% 0.19% "/wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe"

### Similar long running action scheduler related requests ones: 
6.159 seconds:
[24/May/2024:00:03:17 +0000] "POST /wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe" 200 0 - 366682 425283 **6.159** 14680064 1.46% 0.65% "/wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe"

5.130 seconds:
[24/May/2024:00:03:23 +0000] "POST /wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe" 200 0 - 366682 425281 **5.130** 14680064 1.36% 0.78% "/wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe"

5.205 seconds:
[24/May/2024:00:05:19 +0000] "POST /wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe" 200 0 - 366682 425397 **5.205** 14680064 2.31% 0.38% "/wp-admin/admin-ajax.php?action=as_async_request_queue_runner&nonce=d3ece64bfe"

Basically all other GET and POST request in the PHP access log file having a PHP duration of 0.100-0.200 (100-200 milliseconds) whereas this action scheduler related log is always around ~5.0+ seconds. This appears associated with the Action Schedular and WP Rocket appears to be the only thing using Action Schedular library on this site, woocommerce is not installed. 

Edit: WPForms is also using action scheduler library, investigating. 

Compare to other POST requests that are fast:
0.096s
[24/May/2024:00:06:00 +0000] "POST /index.php" 201 47 - 366682 425400 **0.096** 16777216 83.11% 10.39% "/wp-json/wordpress-popular-posts/v2/views/61578"

0.189s
[24/May/2024:00:06:57 +0000] "POST /wp-cron.php?doing_wp_cron=1716509217.1633090972900390625000" 200 0 - 366682 425457 **0.189** 14680064 63.48% 21.16% "/wp-cron.php?doing_wp_cron=1716509217.1633090972900390625000"

0.082s 
[24/May/2024:00:09:57 +0000] "POST /wp-comments-post.php" 200 192 - 366682 425618 0.082 14680064 85.16% 12.17% "/wp-comments-post.php"

**To Reproduce**
Edit your /etc/php-(version)/fpm-pools.d/  conf file or similar for your distro 
Add "%{seconds}d"  somewhere in  "access.format =" line of conf file  so it will start logging PHP duration if not present
Observe new log with duration logging and find admin-ajax.php?action=as_async_request_queue_runner to see duration of these requests compared to others. 

**Expected behavior**
Not take seconds, but rather hundreds of milliseconds. Are others seeing long running php processes for the action scheduler? Maybe this in normal but it doesn't seem like it should be. 

**Screenshots**

**Additional context**

**Acceptance Criteria (for WP Media team use only)**
Clear instructions for developers, to be added before the grooming

**Comment by piotrbak** [comment]
Hello @BradFD queue runner is processing tasks that are in the queue with that process. It sounds like an expected behaviour.

**Comment by camilamadronero** [comment]
**Related case:** https://secure.helpscout.net/conversation/2954359065
**Issues:** the requests were getting a 499 error
